mlrun/projects/pipelines.py

```diff
@@ -528,11 +523,15 @@ def run(
         secrets=None,
         artifact_path=None,
         namespace=None,
+        source=None,
     ) -> _PipelineRunStatus:
         pipeline_context.set(project, workflow_spec)
         workflow_handler = _PipelineRunner._get_handler(
             workflow_handler, workflow_spec, project, secrets
         )
+        if source and not workflow_spec.run_local:
+            logger.info(f"setting project source: {source}")
+            project.set_source(source, pull_at_runtime=True)
```

Comment:
so if i want to run a specific pipeline and pass it a source it becomes the project source and change the default state of the project regarding the `pull_at_runtime`, why? can't we just apply it on the pipeline?

Reply:
The `pull_at_runtime` is an error, removing. See 4. in description.